### ISSUE TITLE
Windows.Devices.Radios may be not work correctly in x86 

### ISSUE THREAD

**Issue by Saberlion** [Microsoft/cppwinrt#47]
My OS is win10 64 bit and the OS version is 14393.0. When I build the code in x64, the program works well. However, when in x86, nothing can be gotten from Windows.Devices.Radios. The code is blow.

```C++
#pragma comment(lib,"windowsapp")
#include "winrt/Windows.Devices.Radios.h"
using namespace winrt;
using namespace  Windows;
using namespace Windows::Devices::Radios;
using namespace Windows::Foundation;
using namespace Windows::Foundation::Collections;

void test()
{
	auto list = Radio::GetRadiosAsync();
	while (list.Status() != AsyncStatus::Completed)
	{
		Sleep(100);
	}
	IVectorView<Radio> radios = list.GetResults();
	int n = radios.Size();
	auto radioItem = radios.First();

	for (int i = 0; i < radios.Size(); i++)
	{
		Radio r = radioItem.Current();
		radioItem.MoveNext();
	}
	return;
}
int main()
{
	initialize();
	test();
	return 0;
}
```

**Comment by kennykerr** [comment]
Here's a simpler way to enumerate the radios:

```C++
IVectorView<Radio> radios = Radio::GetRadiosAsync().get();

for (Radio const & radio : radios)
{
    printf("%ls\n", radio.Name().c_str());
}
```

Can you be more specific about what may not be working?

**Comment by Saberlion** [comment]
Thank you for your reply. 
When I build the code in x64 and run it, the program can get two radios, the wlan and the bluetooth. However, when in x86, the program can get neither of the radios. It is the same result when I test the program in another PC.

**Comment by kennykerr** [comment]
I found a machine with two radio devices (my desktop has none). On x64 GetRadiosAsync produces a collection with two elements, but on x86 it returns an empty collection. This seems unrelated to C++/WinRT but I will check with the team responsible for this API.

**Comment by Saberlion** [comment]
It seems that something goes wrong with the API on x86.
Thank you.

**Comment by marcusl** [comment]
This is still an issue. I'm running Win 10 1703!

**Comment by kennykerr** [comment]
Sorry folks, since the issue was closed I did not have a reminder to report back on this question. It turns out this is "by design". 😉 

The issue is that the WinRT API tries to load registered media managers to listen for change notifications. These media managers are COM servers and are only expected to be created in the native OS architecture. The bottom line is that Windows does not register or ship a Wow64 version. The reason it works in an app container is because in that context the WinRT API makes the call through a broker process, which runs natively using the OS architecture.
